File: apps/api_server/src/models/project_template.ts
export interface ProjectTemplateStep {
  id: string;
  templateId: string;
  title: string;
  offsetDays: number;
  offsetDescription: string | null;
  sortOrder: number;
  assigneeId: number | null;
  assigneeName: string | null;
}

export interface ProjectTemplate {
  id: string;
  name: string;
  description: string | null;
  anchorType: string;
  ownerId: number | null;
  createdAt: string;
  steps: ProjectTemplateStep[];
}

export interface CreateProjectTemplateDto {
  name: string;
  description?: string | null;
  anchorType?: string;
  ownerId?: number | null;
}

export interface UpdateProjectTemplateDto {
  name?: string;
  description?: string | null;
  ownerId?: number | null;
}

export interface CreateStepDto {
  title: string;
  offsetDays: number;
  offsetDescription?: string | null;
  sortOrder?: number;
  assigneeId?: number | null;
}

File: apps/desktop_flutter/lib/features/projects/models/project_template.dart
import 'project_template_step.dart';

class ProjectTemplate {
  ProjectTemplate({
    required this.id,
    required this.name,
    required this.anchorType,
    required this.createdAt,
    required this.steps,
    this.description,
  });

  factory ProjectTemplate.fromJson(Map<String, dynamic> json) {
    final stepList = (json['steps'] as List<dynamic>? ?? [])
        .map((s) => ProjectTemplateStep.fromJson(s as Map<String, dynamic>))
        .toList();
    return ProjectTemplate(
      id: json['id'] as String,
      name: json['name'] as String,
      description: json['description'] as String?,
      anchorType: json['anchorType'] as String? ?? 'date',
      createdAt: json['createdAt'] as String,
      steps: stepList,
    );
  }

  final String id;
  final String name;
  final String? description;
  final String anchorType;
  final String createdAt;
  final List<ProjectTemplateStep> steps;
}
